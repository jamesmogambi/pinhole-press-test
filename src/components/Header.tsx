import React from "react";

const Header: React.FC<{ states: IState[] }> = ({ states }) => {
  return (
    <React.Fragment>
      <div className="header">
        <div className="container shadow">
          <span className="title">PRODUCTS</span>
          <div className="items">
            <span className="item">Category</span>
            <span className="item">Sub-Category</span>
          </div>
        </div>
        <div className="container">
          <span className="title title-state">STATE</span>
          <div className="items">
            {states.map((s, i) => (
              <span key={i} className="item">
                {s.state}
              </span>
            ))}
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default Header;
