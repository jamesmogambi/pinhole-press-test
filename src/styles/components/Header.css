.header {
  background: var(--dark-blue-color);
  height: 105px;
  display: flex;
  padding-top: 7px;
}

.title {
  color: var(--white-color);
  font-weight: bold;
  padding-left: 10px;
}

.item {
  color: rgba(250, 235, 215, 0.774);
  width: 120px;
  font-size: 15px;
  padding: 10px;
}

.items {
  margin-top: 20px;
  display: flex;
  flex-direction: row;
}

.title-state {
  position: relative;
  left: 8%;
}
