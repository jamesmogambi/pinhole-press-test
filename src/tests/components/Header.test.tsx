import { shallow } from "enzyme";
import Header from "../../components/Header";

test("should correctly render Header with states", () => {
  const states = [
    { state: "Alabama", city: "Decatur" },
    { state: "Arizona", city: "Gilbert" },
    { state: "Arkansas", city: "Fayetteville" },
    { state: "California", city: "Los Angeles" },
    { state: "Colorado", city: "Aurora" },
    { state: "Connecticut", city: "Fairfield" },
  ];

  let wrapper = shallow(<Header states={states} />);
  expect(wrapper).toMatchSnapshot();
});
