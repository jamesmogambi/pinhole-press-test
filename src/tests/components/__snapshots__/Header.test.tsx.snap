// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Header with states 1`] = `
<Fragment>
  <div
    className="header"
  >
    <div
      className="container shadow"
    >
      <span
        className="title"
      >
        PRODUCTS
      </span>
      <div
        className="items"
      >
        <span
          className="item"
        >
          Category
        </span>
        <span
          className="item"
        >
          Sub-Category
        </span>
      </div>
    </div>
    <div
      className="container"
    >
      <span
        className="title title-state"
      >
        STATE
      </span>
      <div
        className="items"
      >
        <span
          className="item"
          key="0"
        >
          Alabama
        </span>
        <span
          className="item"
          key="1"
        >
          Arizona
        </span>
        <span
          className="item"
          key="2"
        >
          Arkansas
        </span>
        <span
          className="item"
          key="3"
        >
          California
        </span>
        <span
          className="item"
          key="4"
        >
          Colorado
        </span>
        <span
          className="item"
          key="5"
        >
          Connecticut
        </span>
      </div>
    </div>
  </div>
</Fragment>
`;
